### DISCUSSION TITLE
New Demo: Connecting nodes by dragging and dropping

### DISCUSSION THREAD

**Discussion by kumilingus** [clientIO/joint#2275] [News]
How to **create a new connection when** the user **drags an element over another element**? How to automatically place a new element next to the destination element and let the user **choose which direction** it should be placed.

<img width="248" alt="image" src="https://github.com/clientIO/joint/assets/3967880/b0eea3d2-93c0-4f63-a29d-186079eb1170">

<img width="138" alt="image" src="https://github.com/clientIO/joint/assets/3967880/06ed68a0-3747-442d-804c-7097c152f12d">

What if a user **drags an element over a link**? The new link can be split into two and the element inserted between them.

<img width="420" alt="image" src="https://github.com/clientIO/joint/assets/3967880/75ca4c9e-4f90-4713-83d2-a04a7978cf51">

  All of this can be **visually aided by SVG previews** to help the user understand what effect each drop action will have. See for yourself in today's [demo](https://codepen.io/jointjs/full/bGQxdbj).

**Comment by raisahmad6** [comment]
can i got this example in react as its in HTML and JS.

**Reply by jamesgeorgewilliams** [reply]
This example is written is JavaScript only, so we don't have a React version. It's possible to integrate JointJS with React, so you could definitely adapt this example to work in your React application.

In the JointJS+ trial package, you can find React versions of our Kitchen Sink and Chatbot applications.

https://www.jointjs.com/demos/kitchen-sink-app
https://www.jointjs.com/demos/chatbot

We also provide our Diagram Index and Tabs demos in React. The source code can be found on github in the links below.

https://www.jointjs.com/demos/diagram-index
https://github.com/clientIO/joint-plus-tutorial-react/tree/mui-tree-view

https://www.jointjs.com/demos/tabs
https://github.com/clientIO/joint-plus-tutorial-react/tree/react-tabs
